benchbuild/source/base.py

```diff
@@ -0,0 +1,152 @@
+"""
+Provide a base interface for downloadable sources.
+"""
+import abc
+import itertools
+import typing as tp
+
+import attr
+import plumbum as pb
+
+from benchbuild.settings import CFG
+
+NestedVariants = tp.Iterable[tp.Tuple[tp.Any, ...]]
+
+
+@attr.s(repr=False)
+class Variant:
```

Comment:
Can you please add a bit of documentation for Variant and the other new things. As I currently understand it it warps together the sources and the version ob a project. Is this designed to be extended in the futuer?

Reply:
Added comment in merged file.